holidays/financial/ny_stock_exchange.py

```
@@ -55,88 +60,185 @@ def __init__(self, *args, **kwargs):
+            1947: ((MAY, 31), (SEP, 27)),
+            1948: ((MAY, 29), (SEP, 25)),
+            1949: ((MAY, 28), (SEP, 24)),
+            1950: ((JUN, 3), (SEP, 30)),
+            1951: ((JUN, 2), (SEP, 29)),
+            1952: ((MAY, 31), (SEP, 27)),
+        }
+        if self._year in closed_sat_date_ranges:
+            start, end = closed_sat_date_ranges[self._year]
+            self._populate_ranged_holidays("Closed Saturday", start, end, 7)
+
+    def _populate_half_day_holidays(self):
```

Ruff (0.14.7)

Missing return type annotation for private function `_populate_half_day_holidays`

Add return type annotation: `None`

(ANN202)